account_payment_term_partner_holiday/models/res_partner.py

```diff
@@ -21,7 +21,11 @@ def is_date_in_holiday(self, date):
             date = fields.Date.from_string(date)
         res = self.holiday_ids.filtered(
             lambda x: int(x.month_from) <= date.month and int(x.month_to) >= date.month
-            and int(x.day_from) <= date.day and int(x.day_to) >= date.day
+            and int(x.day_from) <= date.day
+            and (
+                (int(x.day_to) >= date.day and int(x.month_to) >= date.month)
+                or (int(x.day_to) <= date.day and int(x.month_to) > date.month)
+            )
```

Comment:
Instead of all this, can't you do something like:
1. Take the current `year` from `date` 
1. Generate a start date for the holiday period for that year from `holiday_ids` 
3. Do the same for an end date
3. Check if `date` is between them with normal `Date` object comparison

Reply:
Maybe something like this could work @victoralmau:

```python
def is_date_in_holiday(self, date):
    if isinstance(date, str):
        date = fields.Date.from_string(date)
    for holiday in self.holiday_ids:
        holiday_start_date = self._generate_field_date(date.year, holiday.month_from, holiday.day_from)
        # If the user selected a date of the next year (e.g. vacations from December to January), compute the correct date
        holiday_end_date = self._generate_field_date(year_to, holiday.month_to, holiday.day_to) if (holiday.month_to > holiday.month_from or (holiday.month_to == holiday_month_from and holiday.day_to > holiday.day_from)) else self._generate_field_date(date.year + 1, holiday.month_to, holiday.day_to)
    	if date >= holiday_start_date and date <= holiday_end_date:
            return [holiday_start_date,holiday_end_date]
    return False

def _generate_field_date(year, month, day):
    try:
        return fields.Date.from_string("%s-%s-%s" % (year, month, day))
    except ValueError:
        # When the user selects a date that does not exist, assume the last day for that month
        return fields.Date.from_string("%s-%s-%s" % (year if month < 12 else year + 1, month + 1 if month < 12 else 1, 1)) - timedelta(day=1)
```

Reply:
ping @pedrobaeza what do you think about these idea to change function?

Reply:
> Why substracting one day?

This is a trick to get the last day of the current month: Go to the first day of the following month and subtract 1.

Reply:
I think you can get that with `calendar` from dateutils, but OK as well.

Reply:
Indeed, I see there are a couple of ways of doing this:

1. Use `datetutil.relativedelta`. Is is similar to `timedelta` but, if you add 31 days to any day of the month, you willl get the last day of that month: https://dateutil.readthedocs.io/en/stable/relativedelta.html
2. Use `calendar.monthrange` that returns the number of days in a month, so you can get if from there: https://docs.python.org/3/library/calendar.html#calendar.monthrange

@victoralmau it is up to you if you want to change it or not. Any of the solutions look good to me 👍

Reply:
I have changed to `calendar.monthrange`
According to these change, it's a good idea to change `_generate_field_date()` too to simplify?

```
def _generate_field_date(self, year, month, day):
    # When the user selects a date that does not exist, assume the last day
    # for that month
    return fields.Date.from_string(
        "%s-%s-%s" % (year, month, max(calendar.monthrange(year, month)))
    )
```

Reply:
It should be `min` and include the end day.

Reply:
I would maybe keep the comment or a docstring in `_generate_field_date` to explain what you are doing, otherwise LGTM :+1: